spark/src/main/scala/org/apache/spark/sql/delta/DeltaParquetFileFormat.scala

```diff
@@ -374,25 +412,37 @@ case class DeltaParquetFileFormat(
           // tables. https://github.com/delta-io/delta/issues/2246
           val newRow = columnarRow.copy();
           isRowDeletedColumn.foreach { columnMetadata =>
-            rowIndexFilter.get.materializeIntoVector(rowIndex, rowIndex + 1, tempVector)
+            if (useMetadataRowIndex) {
+              val rowIndex = columnarRow.getLong(rowIndexColumn.get.index)
+              rowIndexFilter.get.materializeSingleRowWithRowIndex(rowIndex, tempVector)
+            } else {
+              rowIndexFilter.get.materializeIntoVector(rowIndex, rowIndex + 1, tempVector)
```

Comment:
can we use `rowIndexFilter.get.materializeSingleRowWithRowIndex(rowIndex, tempVector)` here as well? Also same comment below.